app/uk/gov/hmrc/thirdpartydeveloperfrontend/service/ApplicationService.scala

```diff
@@ -207,6 +207,21 @@ class ApplicationService @Inject() (
     )
   }
 
+  def requestProductonApplicationNameChange(application: Application, newApplicationName: String, requesterName: String, requesterEmail: String)(implicit hc: HeaderCarrier) = {
+
+    def createDeskproTicket(application: Application, newApplicationName: String, requesterName: String, requesterEmail: String) = {
+      val previousAppName = application.name
+      val appId = application.id
+
+      DeskproTicket.createForRequestChangeOfProductionApplicationName(requesterName, requesterEmail, previousAppName, newApplicationName, appId)
+    }
+
+    val ticket = createDeskproTicket(application, newApplicationName, requesterName, requesterEmail)
+    for {
+      ticketResponse <- deskproConnector.createTicket(ticket)
```

Comment:
is the for...yield needed here, could we just return deskproConnector.createTicket(ticket) ?

Reply:
Fixed